python/HeaderService/hslib_salobj.py

```diff
@@ -60,28 +60,29 @@ def __init__(self, **keys):
         # Define the callbacks for start/end
         self.define_evt_callbacks()
 
-        # Initialize the timeout task
-        self.end_evt_timeout_task = salobj.make_done_future()
-
     async def close_tasks(self):
         """Close tasks on super and evt timeout"""
         await super().close_tasks()
-        self.end_evt_timeout_task.cancel()
+        # Loop over all task and cancel all of them
+        for imageName in self.end_evt_timeout_task:
+            self.end_evt_timeout_task[imageName].cancel()
 
-    async def end_evt_timeout(self, timeout):
+    async def end_evt_timeout(self, imageName, timeout):
         """Timeout timer for end event telemetry callback"""
         await asyncio.sleep(timeout)
         self.log.info(f"{self.name_end} not seen in {timeout} seconds; giving up")
         # Send the timeout warning using the salobj log
-        self.log.warning(f"Timeout while waiting for {self.name_end} Event")
-        self.clean()
+        self.log.warning(f"Timeout while waiting for {imageName} {self.name_end} Event")
+        self.clean(imageName)
 
     def report_summary_state(self):
         """Call to report_summary_state and LOGGER"""
         super().report_summary_state()
         self.log.info(f"Current state is: {self.summary_state.name}")
         if self.summary_state != salobj.State.ENABLED:
-            self.end_evt_timeout_task.cancel()
+            # Loop over all task and cancel all of them
+            for imageName in self.end_evt_timeout_task:
```

Comment:
You'll need a lock around this code, since you're messing with the end_evt_timeout_task list and since you released the lock in the calling method before this method was called.